File: ruby/matematica/3096.rb
def kamenetsky(n)
  return 0 if n < 0
  return 1 if n <= 1

  x = ((n * Math.log10(n/Math::E)) +
      (Math.log10(2*Math::PI*n) / 2.0))

  return x.floor + 1
end

n = gets.chomp.to_f
puts kamenetsky(n)

File: elixir/matematica/3096.ex
defmodule M do
    def kamenetsky(0) do
        0
    end
    def kamenetsky(1) do
        1
    end
    def kamenetsky(n) do
      x = (n * :math.log10(n / :math.exp(1.0)) +
            ( :math.log10(2.0 * :math.pi * n) / 2.0) )
      :math.floor(x) + 1
    end
    
    def main do
        {n,_} = IO.gets("") |> String.trim |> Float.parse
        result = kamenetsky(n)
        IO.puts trunc(result)
    end
end

M.main()
